File: materials/clean_code/simple_example/Python/OOP/iteratively/good_example_oop_iteratively.py
class number_properties():
    """docstring for number_properties."""

    def get_fibonacci_number(self, position_of_number):

        position_of_number += 1

        start_point_value = 0
        final_value = 1

        if position_of_number == 0:
            return 0

        for iterator in range(2, position_of_number, 1):

            intermediate_value = start_point_value + final_value
            start_point_value = final_value
            final_value = intermediate_value

        return final_value


    def get_factorial_number(self, position_of_number):

        position_of_number += 1
        result = 1

        for iterator in range(1, position_of_number, 1):
            result *=  iterator

        return result


class test_cases():
    """docstring for test_cases."""

    def assert_fibonacci(self, properties):
        if properties.get_fibonacci_number(properties, 7) == 13:
            print("Fibonacci test passed")


    def assert_factorial(self, properties):
        if properties.get_factorial_number(properties, 3) == 6:
            print("Factorial test passed")

properties = number_properties
cases = test_cases

cases.assert_fibonacci(cases, properties)
cases.assert_factorial(cases, properties)

File: materials/clean_code/simple_example/Java/iteratively/good_example_oop_iteratively.java
class NumberProperties {

  public static int GetFibonacciNumber (int PositionOfNumber) {

    int StartPointValue = 0;
    int FinalValue = 1;
    int IntermediateValue;

    if (PositionOfNumber == 0)
      return 0;

    for (int iterator = 2; iterator <= PositionOfNumber; iterator++) {

      IntermediateValue = StartPointValue + FinalValue;
      StartPointValue = FinalValue;
      FinalValue = IntermediateValue;
    }

    return FinalValue;
  }

  public static int GetFactorialNumber (int PositionOfNumber) {

    int Result = 1;

    for (int iterator = 1; iterator <= PositionOfNumber; iterator++)
      Result *= iterator;

    return Result;
  }
}

class TestCases {

  public static void AssertFibonacci (NumberProperties Properties) {

    if (Properties.GetFibonacciNumber(7) == 13)
      System.out.println("Fibonacci test passed");
  }

  public static void AssertFactorial (NumberProperties Properties) {

    if (Properties.GetFactorialNumber(3) == 6)
      System.out.println("Factorial test passed");
  }
}

class Main {

  public static void main (String[] args) {

    NumberProperties Properties = new NumberProperties();
    TestCases Cases = new TestCases();

    Cases.AssertFibonacci(Properties);
    Cases.AssertFactorial(Properties);
  }
}
